src/openlcb/TractionCvSpace.cxx

```diff
@@ -330,7 +330,7 @@ void TractionCvSpace::send(Buffer<dcc::RailcomHubData> *b, unsigned priority)
     if (errorCode_ != ERROR_PENDING)
         return;
     const dcc::Feedback &f = *b->data();
-    if (f.feedbackKey != reinterpret_cast<size_t>(this) || f.channel == 0xff)
+    if (f.feedbackKey != (uint32_t)(reinterpret_cast<size_t>(this)) || f.channel == 0xff)
```

Comment:
It if it is a pointer, why would you not use type uintptr_t?  That would be portable between 32 and 64 bit.

Reply:
feedbackKey is uint32_t. This is a member in the packet_t structure which is not internal to our application; this struct is used to communicate with the DCC driver. We are not free to modify the size of this struct or the members.

Reply:
I'm a bit concerned about this on 64-bit systems.  As soon as you cast 64-bit pointer to a uint32_t, it is no longer guaranteed to be unique.  On a big endian system, it might be less likely to be unique.  Maybe it would be better to change the type of feedbackKey to unit64_t.  What is the reason that feedbackKey cannot be unitptr_t?  Is it because the instance can be passed between host architectures (PRU vs. ARM for example)?

Reply:
yes it gets passed to the PRU

Reply:
ok I switched everything to uintptr_t.